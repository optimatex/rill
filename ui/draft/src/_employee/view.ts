export {
  React,
  useState,
  useCallback,
  css,
  styled,
  createGlobalStyle,
} from '@rill/view';
