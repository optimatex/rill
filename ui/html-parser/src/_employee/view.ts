export { React, RC } from '@rill/view';
