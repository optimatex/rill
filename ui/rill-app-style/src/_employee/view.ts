export { createGlobalStyle, css } from '@rill/view';
