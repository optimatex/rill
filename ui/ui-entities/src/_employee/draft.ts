export { TextEditor } from '@rill/draft';
