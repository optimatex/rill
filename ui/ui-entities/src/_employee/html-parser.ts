export { htmlParser } from '@rill/html-parser';
