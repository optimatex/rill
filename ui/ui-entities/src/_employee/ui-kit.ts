export {
  // Layout
  Row,
  ListContainer,
  // Card
  Card,
  Post,
  Panel,
  // Menu
  MenuItemLink,
  // Typography
  Paragraph,
  Name,
  // Image
  Image,
  ImageCropper,
  ImagesLayout,
  // Video
  YoutubeVideos,
  // Content
  RichContentContainer,
} from '@rill/ui-kit';
