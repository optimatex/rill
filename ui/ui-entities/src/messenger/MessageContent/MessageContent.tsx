import { React } from '_employee/view';
import { htmlParser } from '_employee/html-parser';
import { RichContentContainer } from '_employee/ui-kit';

import renders from './contentRenderParams';

interface IProps {
  html: string;
}

const MessageContent = (props: IProps) => (
  <RichContentContainer>{htmlParser(props.html, renders)}</RichContentContainer>
);

export default MessageContent;
