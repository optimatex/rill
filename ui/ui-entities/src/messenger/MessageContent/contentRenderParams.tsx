import { React, RC } from '_employee/view';
import { Paragraph } from '_employee/ui-kit';

const renderParagraph = (children: RC, index: number) => (
  <Paragraph index={index} key={index}>
    {children}
  </Paragraph>
);

const renderLink = (url: string, children: RC) => <a key={url}>{children}</a>;

const renders = { renderParagraph, renderLink };

export default renders;
