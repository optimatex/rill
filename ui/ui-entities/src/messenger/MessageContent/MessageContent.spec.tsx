import { React } from '_employee/view';
import { render, mockHtmlString_1 } from 'test';
import MessageContent from '.';

describe('MessageContent', () => {
  it('should render without errors', () => {
    const { container } = render(<MessageContent html={mockHtmlString_1} />);

    expect(container).toMatchSnapshot();
  });
});
