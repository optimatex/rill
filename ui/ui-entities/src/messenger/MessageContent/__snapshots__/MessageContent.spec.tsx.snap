// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageContent should render without errors 1`] = `
.c1 {
  font-size: 8px;
  font-family: roboto;
  font-weight: bolder;
}

.c0 p {
  padding: 0;
  margin: 0;
}

<div>
  <div
    class="c0"
  >
    <p
      class="c1"
    >
      I've been a QA engineer with comp.sci background working in many places building quality tools for over 11 years. Last year during the pandemic most of my company was fired, including myself. So, I decided to apply for a web developer role at Napster. This video was literally all I used to do the technical exam. I am now finishing my probation period! Thanks, mate!
    </p>
  </div>
</div>
`;
